core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -1960,9 +1962,16 @@ class KafkaConfig private(doLog: Boolean, val props: java.util.Map[_, _], dynami
   }
 
   def listenerSecurityProtocolMap: Map[ListenerName, SecurityProtocol] = {
-    getMap(KafkaConfig.ListenerSecurityProtocolMapProp, getString(KafkaConfig.ListenerSecurityProtocolMapProp))
+    val mapValue = getMap(KafkaConfig.ListenerSecurityProtocolMapProp, getString(KafkaConfig.ListenerSecurityProtocolMapProp))
       .map { case (listenerName, protocolName) =>
-      ListenerName.normalised(listenerName) -> getSecurityProtocol(protocolName, KafkaConfig.ListenerSecurityProtocolMapProp)
+        ListenerName.normalised(listenerName) -> getSecurityProtocol(protocolName, KafkaConfig.ListenerSecurityProtocolMapProp)
+      }
+    if (usesSelfManagedQuorum && !originals.containsKey(ListenerSecurityProtocolMapProp)) {
+      // Nothing was specified explicitly, so we are using the default value;
+      // therefore, since we are using KRaft, add the CONTROLLER:PLAINTEXT mapping
+      mapValue ++ Map(new ListenerName("CONTROLLER") -> SecurityProtocol.PLAINTEXT)
```

Comment:
Why do we hard-code the "CONTROLLER" listener. Shouldn't we be using the values in `controller.listener.names`?

Reply:
> Why do we hard-code the "CONTROLLER" listener. Shouldn't we be using the values in controller.listener.names?

Hmm, that's an interesting point.  We had discussed offline about adding the `CONTROLLER:PLAINTEXT` default mapping in KRaft mode, but we never considered that maybe we should default all specified controller listeners to PLAINTEXT when there is no listener security protocol map explicitly provided.  That does seem more reasonable at first glance.  @cmccabe thoughts on this?

Reply:
What if someone names their controller listener SSL ?

I suppose you could say if there the controller listener name is not the name of a security protocol, and there is no map defined, then default the listener to PLAINTEXT.

Reply:
> I suppose you could say if there the controller listener name is not the name of a security protocol, and there is no map defined, then default the listener to PLAINTEXT

Yes, definitely.  I'll go ahead and make this change.

Reply:
The default mapping here doesn't do us any good unless it corresponds to one of the values in controller.listener.names, no?

> What if someone names their controller listener SSL ?

Interestingly, the logic we use today allows for this explicitly. Here is how we currently determine the security protocol in `BrokerToControllerChannelManager`. 

```scala
    val controllerSecurityProtocol = config.listenerSecurityProtocolMap.getOrElse(controllerListenerName, SecurityProtocol.forName(controllerListenerName.value()))
```